jni/example/lib/main.dart

```diff
@@ -9,104 +9,102 @@ import 'package:flutter/material.dart';
 import 'dart:io';
 import 'dart:ffi';
 
-import 'package:ffi/ffi.dart';
 import 'package:jni/jni.dart';
-import 'package:jni/jni_object.dart';
 
-late Jni jni;
-
-String localToJavaString(int n) {
-  final jniEnv = jni.getEnv();
+// An example of calling JNI methods using low level primitives.
+String toJavaStringUsingEnv(int n) {
+  // GlobalJniEnv is a thin abstraction over JNIEnv in JNI C API.
+  // For a more ergonomic API for common use cases of calling methods and
+  // accessing fields, see next examples using JniObject and JniClass.
+  final env = Jni.env;
   final arena = Arena();
-  final cls = jniEnv.FindClass("java/lang/String".toNativeChars(arena));
-  final mId = jniEnv.GetStaticMethodID(cls, "valueOf".toNativeChars(),
+  final cls = env.FindClass("java/lang/String".toNativeChars(arena));
+  final mId = env.GetStaticMethodID(cls, "valueOf".toNativeChars(),
       "(I)Ljava/lang/String;".toNativeChars(arena));
   final i = arena<JValue>();
   i.ref.i = n;
-  final res = jniEnv.CallStaticObjectMethodA(cls, mId, i);
-  final str = jniEnv.asDartString(res);
-  jniEnv.deleteAllLocalRefs([res, cls]);
+  final res = env.CallStaticObjectMethodA(cls, mId, i);
+  final str = env.asDartString(res);
+  env.deleteAllRefs([res, cls]);
   arena.releaseAll();
   return str;
 }
 
-int random(int n) {
+int randomUsingEnv(int n) {
   final arena = Arena();
-  final jniEnv = jni.getEnv();
-  final randomCls = jniEnv.FindClass("java/util/Random".toNativeChars(arena));
-  final ctor = jniEnv.GetMethodID(
+  final env = Jni.env;
+  final randomCls = env.FindClass("java/util/Random".toNativeChars(arena));
+  final ctor = env.GetMethodID(
       randomCls, "<init>".toNativeChars(arena), "()V".toNativeChars(arena));
```

Comment:
How are the method signatures constructed?

Can we construct `"()V"`, `"(I)I"`, `"()D"`, and ones with parameter types from a helper data structure? Would that be useful to have for users directly interacting with package:jni, instead of jnigen? (I presume we generate these in jnigen.) Feel free to file a new issue for this rather than address this in this PR.

Reply:
I replied to the issue.